### ISSUE TITLE
Error on new DID create operation

### ISSUE THREAD

**Issue by ALRubinger** [TBD54566975/ssi-service#68]
From https://forums.tbd.website/t/tutorials-and-examples-system/89/6:

```
%> curl -X PUT -d '{"keyType":"Ed25519"}' localhost:8080/v1/dids/key

{"error":"invalid create DID request"}
```

@decentralgabe noted he found a regression; adding here for tracking.

**Comment by decentralgabe** [comment]
It is a regression cause by https://github.com/TBD54566975/ssi-service/pull/60/files#diff-547116889535d0344e07cd1360235b506d311930cbe7bedae28575fd2961f1c1R32 which reads the request buffer, leaving it empty.

A fix will be to read the buffer and replace it, or to copy the buffer before reading it.

Regression not caught in testing, since CI does not run open telemetry. Will be caught once integration tests are up in #13

**Comment by ALRubinger** [comment]
Hmm, I think I'm still running into this. Ran:

```
% git pull upstream main
...
Already up to date.

% mage cbt
...

% cd build && docker compose up
...
```

Readiness probe still working:
```
% curl localhost:8080/readiness
{"status":{"status":"ready","message":"all service ready"},"serviceStatuses":{"credential":{"status":"ready"},"did":{"status":"ready"},"schema":{"status":"ready"}}}
```

Same `invalid create DID request` as before, though:
```
% curl -X PUT -d '{"keyType":"Ed25519"}' localhost:8080/v1/dids/key
{"error":"invalid create DID request"}
```

**Comment by frankhinek** [comment]
I'm getting the same error message as @ALRubinger when attempting to create a new DID with [commit 8125f7a](https://github.com/TBD54566975/ssi-service/commit/8125f7a7323572e661c04068a12258a9b93f2b4f).

@decentralgabe It seems to me that the `PeekRequestBody()` function is still reading the buffer and leaving it empty.  [TeeReader](https://pkg.go.dev/io#TeeReader) is reading once from the `io.Reader` implemented by the `body` of http requests and writing to `&buf`, leaving `r.Body` empty.

the io.Reader implemented by the body field of Go http requests, it becomes impossible to read from it again. There is also no builtin way to simply "reset" the body as part of the interface.

https://github.com/TBD54566975/ssi-service/blob/a63d11e96db6c7d908388939c3bcb2de36681449/pkg/server/framework/util.go#L34

I believe I've confirmed this by over-simplifying the request body read in the `Handle()` function of `server.go`

https://github.com/TBD54566975/ssi-service/blob/8125f7a7323572e661c04068a12258a9b93f2b4f/pkg/server/framework/server.go#L67

as follows:

```go
...
import (
...
	"bytes"
	"io"
...
func (s *Server) Handle(method string, path string, handler Handler, mw ...Middleware) {
...
		// init a span, but only if the tracer is initialized
		if s.tracer != nil {
			var span trace.Span
			ctx, span = s.tracer.Start(ctx, path)

			bodyBytes, err := io.ReadAll(r.Body)
			body := string(bodyBytes)
			r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
			// body, err := PeekRequestBody(r.Body)
```

After rebuilding the container images, the create DID request API call succeeds:

```bash
% curl -X PUT -d '{"keyType":"secp256k1"}' localhost:8080/v1/dids/key
{"did":{"@context":"https://www.w3.org/ns/did/v1","id":"did:key:zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK","verificationMethod":[{"id":"#zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK","type":"EcdsaSecp256k1VerificationKey2019","controller":"did:key:zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK","publicKeyBase58":"25yEtupxZPZjLHLBkGfobU7SEzjZUVfTBKk47v8uHQ53b"}],"authentication":[["#zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK"]],"assertionMethod":[["#zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK"]],"keyAgreement":[["#zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK"]],"capabilityDelegation":[["#zQ3shqvPukyS2iCxDXRsAekgeHhogeVkSnDViFfovDUaQcYuK"]]},"privateKeyBase58":"7vEnHGiuo2VFpp"}%
```

@ALRubinger You may already be doing this _off camera_, but make sure you rebuild the `build_web_1` container after pulling in the latest commit(s):

```bash
cd build
docker-compose build && docker-compose up
```

**Comment by frankhinek** [comment]
> Thank you @frankhinek. I did switch out tee to your approach - it's nicer! Fixed here - #71

🍻 @decentralgabe 

Fix confirmed
